google/cloud/bigtable/test_proxy/cbt_test_proxy.cc

```diff
@@ -0,0 +1,392 @@
+// limitations under the License.
+
+#include "google/cloud/bigtable/test_proxy/cbt_test_proxy.h"
+#include "google/cloud/bigtable/cell.h"
+#include "google/cloud/bigtable/idempotent_mutation_policy.h"
+#include "google/cloud/bigtable/mutations.h"
+#include "google/cloud/bigtable/table.h"
+#include "google/cloud/log.h"
+#include "google/cloud/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/time/time.h"
+#include <google/protobuf/util/time_util.h>
+#include <chrono>
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+namespace test_proxy {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+namespace v2 = ::google::bigtable::v2;
+namespace testpb = ::google::bigtable::testproxy;
+
+// Convert `Status` to `grpc::Status`, discarding any `details`.
+::grpc::Status ToGrpcStatus(Status const& status) {
+  return ::grpc::Status(static_cast<grpc::StatusCode>(status.code()),
+                        std::move(status.message()));
+}
+
+// Convert `Status` to `rpc::Status`, discarding any `details`.
+google::rpc::Status ToRpcStatus(Status const& status) {
+  google::rpc::Status ret;
+  ret.set_code(static_cast<int>(status.code()));
+  ret.set_message(status.message().data(), status.message().size());
+  return ret;
+}
+
+// Helper method to convert cbt row to v2 new_row;
+v2::Row ConvertRowToV2(Row const& row) {
+  v2::Row new_row;
+  new_row.set_key(row.row_key());
+
+  // The results from the open source library are presented in a different style
+  // than the results from the V2 API. In order not to hide unexpected client
+  // behavior, we do not reorder the results. This means that the same family
+  // name or column qualifier may appear multiple times.
+  //
+  // We do, however, combine consecutive cells with the same family or column
+  // qualifier where possible.
+  v2::Family* family = nullptr;
+  v2::Column* column = nullptr;
+
+  for (Cell const& cell : row.cells()) {
+    if (family == nullptr || family->name() != cell.family_name()) {
+      family = new_row.add_families();
+      family->set_name(cell.family_name());
+      column = nullptr;
+    }
+    if (column == nullptr || column->qualifier() != cell.column_qualifier()) {
+      column = family->add_columns();
+      column->set_qualifier(cell.column_qualifier());
+    }
+    v2::Cell* new_cell = column->add_cells();
+    new_cell->set_timestamp_micros(cell.timestamp().count());
+    new_cell->set_value(std::move(cell.value()));
+  }
+  return new_row;
+}
+
+}  // namespace
+
+grpc::Status CbtTestProxy::CreateClient(
+    ::grpc::ServerContext*, testpb::CreateClientRequest const* request,
+    testpb::CreateClientResponse*) {
+  if (request->client_id().empty() || request->data_target().empty()) {
+    return grpc::Status(grpc::StatusCode::INVALID_ARGUMENT,
+                        "both `client_id` and `data_target` must be provided.");
+  }
+
+  auto options =
+      Options{}
+          .set<DataEndpointOption>(request->data_target())
```

Comment:
s/DataEndpointOption/EndpointOption/

`DataEndpointOption` was a specific bigtable thing so you could pass the same `options` to the data client and to the admin clients.

We prefer to use the `EndpointOption` now because that is what all of our other 99 libraries use.